### PR TITLE
add Zap for logging and a few helper make targets, fix aggregator tests

### PR DESCRIPTION
**Pull request by vpavlin**
* `Makefile`
  * added `run`, `postgres` and `postgres-clean` make targets, also made `postgres` dependency for `run` and `test` - not sure if that is useful for you, happy to remove it
* replaced `log` with `Zap` (based on `status-go` usage)
* `aggregator_tests` failed for me due to not cleaning all the database stuff. I just added `DROP TABLE` and `DROP INDEX`, but there must be a better way:)

### CODE REVIEW

File: telemetry/aggregator_test.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 func NewMock() *sql.DB {
-	db, err := sql.Open("postgres", "postgres://telemetry:newPassword@127.0.0.1:5432/telemetry_test")
+	db, err := sql.Open("postgres", "postgres://telemetry:newPassword@127.0.0.1:5432/telemetrydb?sslmode=disable")
```
Comment (vpavlin) [inline telemetry/aggregator_test.go:14]: I added `sslmode=disable` as I did not want to mess with certs for the local DB setup and testing. But if that is something we care about, I can see what can be done for the container:)

File: telemetry/server.go [hunk 2]
```diff
@@ -88,65 +96,73 @@ func (s *Server) createReceivedEnvelope(w http.ResponseWriter, r *http.Request)
...
+		err := respondWithError(w, http.StatusBadRequest, "Could not save the envelope")
+		if err != nil {
+			s.logger.Error("failed to respond", zap.Error(err))
+		}
 	}
 
 	err = respondWithJSON(w, http.StatusCreated, receivedEnvelope)
 	if err != nil {
-		log.Println(err)
+		s.logger.Error("failed to respoin", zap.Error(err))
```
Comment (richard-ramos) [inline telemetry/server.go:120]:
```suggestion
		s.logger.Error("failed to respond", zap.Error(err))
```

File: telemetry/server.go [hunk 3]
```diff
@@ -88,65 +96,73 @@ func (s *Server) createReceivedEnvelope(w http.ResponseWriter, r *http.Request)
...
 	defer r.Body.Close()
 
 	err := receivedEnvelope.put(s.DB)
 	if err != nil {
-		log.Println("could not save envelope", err, receivedEnvelope)
+		s.logger.Error("could not save envelope", zap.Error(err), zap.Any("envelope", receivedEnvelope))
+		err := respondWithError(w, http.StatusBadRequest, "Could not save the envelope")
+		if err != nil {
+			s.logger.Error("failed to respond", zap.Error(err))
+		}
```
Comment (richard-ramos) [inline telemetry/server.go:115]: Should we `return` here?

File: telemetry/server.go [hunk 4]
```diff
@@ -11,17 +11,25 @@ import (
...
 
 type Server struct {
 	Router *mux.Router
 	DB     *sql.DB
+	logger *zap.Logger
 }
 
-func NewServer(db *sql.DB) *Server {
+func NewServer(db *sql.DB) (*Server, error) {
+	logger, err := zap.NewDevelopment()
```
Comment (richard-ramos) [inline telemetry/server.go:24]: In cmd/server we are using `log`.
What do you think if the logger is instantiated there, and passed as a parameter here?, i.e:
```go
func NewServer(db *sql.DB, logger *zap.Logger) (*Server, error) {
	logger := log.Named('server')
```

File: telemetry/aggregator.go [hunk 5]
```diff
@@ -2,23 +2,29 @@ package telemetry
...
-	DB *sql.DB
+	DB     *sql.DB
+	logger zap.Logger
 }
 
-func NewAggregator(db *sql.DB) *Aggregator {
-	return &Aggregator{
-		DB: db,
+func NewAggregator(db *sql.DB) (*Aggregator, error) {
+	logger, err := zap.NewDevelopment()
```
Comment (richard-ramos) [inline telemetry/aggregator.go:16]: Probably we should have a single logger instance and pass it down like if it were a context.

**Review by richard-ramos — APPROVED** [review]
PR looks nice!
I left some comments for your consideration :)

**Comment by vpavlin** [comment]
@richard-ramos addressed the comments, thanks for them

**Pull request merged**